src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/common/gui/component/UIBlockStateButton.java

```diff
@@ -141,4 +144,25 @@ public void drawForeground(GuiRenderer renderer, int mouseX, int mouseY, float p
     @Override
     public void drawBackground(GuiRenderer renderer, int mouseX, int mouseY, float partialTick) {
     }
+    
+    public String getBlockName() {
+        return Block.REGISTRY.getNameForObject(this.iBlockState.getBlock()).toString(); 
+    }
+    
+    public String getBlockProperties() {
+        StringBuffer stringBuffer = new StringBuffer();
+        stringBuffer.append("[");
+        UnmodifiableIterator<Entry<IProperty<?>, Comparable<?>>> iPropertyEntry = iBlockState.getProperties().entrySet()
+                .iterator();
+        while (iPropertyEntry.hasNext()) {
```

Comment:
You don't need a full iterator here... just `for (Entry<IProperty<?>, Comparable<?>> entry : iBlockState.getProperties().entrySet())`

Reply:
```
            if (iPropertyEntry.hasNext())
                stringBuffer.append(",");
```
Full iterator here is for those two lines.

Reply:
This is not code that needs very high performance so this would work well too:
```java
        return Blocks.LOG.getDefaultState().getProperties().entrySet().stream()
            .map(e -> e.getKey().getName() + "=" + e.getValue())
            .collect(Collectors.joining(",", "[", "]"));
```